Sources/XcbeautifyLib/JunitReporter.swift

```diff
@@ -21,12 +21,16 @@ package final class JunitReporter {
 
         if FailingTestCaptureGroup.regex.match(string: line) {
             guard let testCase = generateFailingTest(line: line) else { return }
+            // reduce failing retrys, if any
+            components.removePreviousFailingTestsAfter(testCase)
```

Comment:
I don't know much about JUnit but afaik there is no built-in support for the retry-on-failure mechanism so I don't think JUnit is opinionated.

It is acceptable to agree that if one of the retries succeeds, the output should not contain a failure from such a test.

However, I get your point if all the retry fails for different reasons, but should they be reported more than once if it is one test? Should we fail twice because the error is different even if there is just one test failing? or should we make some decisions and grab one?